src/gretel_client/client.py

```diff
@@ -323,20 +360,29 @@ def noop(x):
         stop=tenacity.stop_after_attempt(MAX_RATE_LIMIT_RETRY),
         wait=tenacity.wait_exponential(multiplier=1, min=2, max=10),
     )
-    def _get_records_sync(self, project: str, params: dict, count: int = None):
-        if count is not None:  # pragma: no cover
-            params["count"] = count
+    def _get_records_sync(self, project: str, *args, **kwargs):
+        """Returns a list of records from the streams api
+
+        Args:
+            project: project to stream records from
+            headers: supports custom header params via kwargs
+            params: supports custom query params via kwargs
+        """
         return (
-            self._get(f"streams/records/{project}", params).get(DATA).get(RECORDS, [])
+            self._get(f"streams/records/{project}", **kwargs)
+            .get(DATA)
+            .get(RECORDS, [])
         )
 
     def __iter_records(
         self,
         project: str,
+        entity_stream: str = None,
         position: Optional[str] = None,
         post_process: Callable = None,
```

Comment:
`Optional` here too even though it wasn't part of the PR

Reply:
updated.